modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/EventSubscriber/WebtoolsAnalyticsEventSubscriber.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_webtools_analytics_rules\EventSubscriber;
+
+use Drupal\oe_webtools_analytics\Event\AnalyticsEvent;
+use Drupal\oe_webtools_analytics\AnalyticsEventInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscriptor for the Webtools Analytics event.
+ */
+class WebtoolsAnalyticsEventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Kernel request event handler.
+   *
+   * @param \Drupal\oe_webtools_analytics\AnalyticsEventInterface $event
+   *   Response event.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   *   Thrown if the entity type doesn't exist.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   *   Thrown if the storage handler couldn't be loaded.
```

Comment:
I think it is fine to catch both of these exceptions and convert them into unchecked runtime exceptions. These exceptions are occurring when you are dynamically loading an unknown entity type, but in here we are retrieving the storage of the `webtools_analytics_rule` which is known to exist.

By catching these exceptions and converting them to runtime exceptions we don't require these to be documented all the way up the call stack, but in the very unlikely event that the storage won't be available then the runtime exception is still thrown and can be logged.

Reply:
Please let me know if the fix I did is what you had in mind :)

Reply:
Yeah, almost! I would be most explicit in which exceptions are being caught, and also document why we are converting them into unchecked runtime exceptions. I will edit this part so you can see what I mean.